### PR TITLE
Update compathelper CI script

### PR DESCRIPTION
**Pull request by exaexa**
a more complete version of hotfix #656 (refreshes the whole CompatHelper script)

### CODE REVIEW

File: .github/workflows/CompatHelper.yml [hunk 1]
```diff
@@ -3,16 +3,35 @@ on:
...
     - cron: 0 0 * * *
   workflow_dispatch:
+permissions:
+  contents: write
+  pull-requests: write
 jobs:
   CompatHelper:
     runs-on: ubuntu-latest
     steps:
+      - name: Check if Julia is already available in the PATH
```
Comment (laurentheirendt) [inline .github/workflows/CompatHelper.yml:13]: Why are these here?

Comment (exaexa) [reply .github/workflows/CompatHelper.yml:13]: it is a clone of the officially recommended CH script:
https://github.com/JuliaRegistries/CompatHelper.jl/blob/master/.github/workflows/CompatHelper.yml

Comment (laurentheirendt) [reply .github/workflows/CompatHelper.yml:13]: Are they from a previous hot fix?

Comment (exaexa) [reply .github/workflows/CompatHelper.yml:13]: (apparently github may provide some julia in base images)

Comment (laurentheirendt) [reply .github/workflows/CompatHelper.yml:13]: okay, but then this will go only later to `master` then ....

**Review by laurentheirendt — APPROVED** [review]
